doc/man7/property.pod

```diff
@@ -0,0 +1,200 @@
+two such strings separated by periods.
+
+=head2 Properties
+
+A property is a I<name=value> pair.
+A property definition is a sequence of comma separated properties.
+There can be any number of properties in a definition.
+For example: "" defines a null property definition; "my.foo=bar" defines
+a property named I<my.foo> which has a string value I<bar> and
+"iteration.count=3" defines a property named I<iteration.count> which has a
+numeric value of I<3>.
+The full syntax for property definitions appears below.
+
+=head2 Implementations
+
+Each implementation of an algorithm can define any number of properties it has.
+For example the default provider defines the property I<default=yes> for all
+of its algorithms.
+Likewise, the FIPS provider defines I<FIPS=yes> and the legacy provider
+defines I<legacy=yes> for all of their algorithms.
+
+=head2 Queries
+
+A property query clause is a single conditional test.  For example, "FIPS=yes",
+"default!=yes" or "?iteration.count!=3".
+The first two represent mandatory clauses, such clauses B<must> match for any
+algorithm to even be in consideration.
+The third clause represents an optional clause, the more of these that match
+the better but not matching one does not rule out using an implementation.
+A query is a sequence of comma separated query clauses.
+The full syntax for property queries appears below.
+
+=head2 Lookups
+
+When an algorithm is looked up a property query is used to determine
+the best matching algorithm.
+All mandatory query clauses B<must> be present and the implementation that
+additionally has the most matching optional query clauses, will be used.
+In the case of tied results, an indeterminate implementation that matches the
+mandatory clauses, will be used.
+
+=head2 Shortcut
+
+In order to permit a more concise expression of Boolean properties, there
+is one short cut: a property name alone (e.g. "default") is
+exactly equivalent to "default=yes" in both definitions and queries.
+
+=head1 SYNTAX IN EBNF
+
+ Definition
+       ::= PropertyName ( '=' Value )? ( ',' PropertyName ( '=' Value )? )*
+ Query          ::= PropertyQuery ( ',' PropertyQuery )*
+ PropertyQuery  ::= '-'? PropertyName
+                  | '?' ( PropertyName (( '=' | '!=' ) Value)?)
+ Value          ::= NumberLiteral | StringLiteral
+ StringLiteral  ::= QuotedString | UnquotedString
+ QuotedString   ::= '"' [^"]* '"' | "'" [^']* "'"
+ UnquotedString ::= [^{space},]+
+ NumberLiteral  ::= '0' ( [0-7]* | 'x' [0-9A-Fa-f]+ ) | '-'? [1-9] [0-9]+
+ PropertyName   ::= [A-Z] [A-Z0-9_]* ( '.' [A-Z] [A-Z0-9_]* )*
+
+=head2 Railroad Diagrams
+
+The syntax is more easily understood using railroad diagrams.  For each rule,
+start at the B<E<gt>E<gt>> symbol and continue until the B<E<gt>E<lt>> symbol.
+If there is a conflict between the railroad diagrams and the eBNF syntax, the
+latter has precedence.
+
+=head3 Definition
+
+     /---- ',' ---------------------------------------\
```

Comment:
Are these diagrams that useful?

Reply:
I'm thinking the EBNF is enough.

Reply:
I can add the eBNF rules.  They aren't as obvious as the railroad diagrams.  At least I don't think they are.

Reply:
This looks like the remaining outstanding point.

To railroad or not to railroad?

I have proper images for these which are more legible but I don't know how they can be included in POD files.

Reply:
BTW: it isn't strict eBNF.  Regular expression syntax is being used which eBNF doesn't allow.  eBNF requires jumping back to English descriptions.  E.g. _all characters except "_.

Reply:
I think the
=head1 SYNTAX IN EBNF section is good enough..

Reply:
POD files can have formatter specific sections.  See https://perldoc.perl.org/perlpod.html (look for `=begin`)

Reply:
Hand crafted ASCII art gone...

What formatter section is suitable for eBNF?

Reply:
There's no special for ebnf, indented (i.e monospace) text is fine

Reply:
Okay, that explains why I couldn't find one :)

Reply:
The railroad diagrams are gone.